### ISSUE TITLE
Export 'ExpressContext' out of the main module. 

### ISSUE THREAD

**Issue by Idono87** [apollographql/apollo-server#3699]
Can we get the `ExpressContext` type exported out of the main module? Currently it's available through the ApolloServer submodule but i prefer not having imports from submodules.

**Comment by abernix** [comment]
In most cases, we try to make sure that types are inferred properly rather than over exporting, however, I can imagine your need makes sense.  That said, since you're encountering the need and you're opening the issue, a small example of your need and usage would be appreciated, so we can make sure the need is well understood.

Furthermore, I suspect a PR would be welcome, but let's take a look at an example of the usage first?

**Comment by Idono87** [comment]
In my case it's to have typings for a custom context with `ExpressContext` as a base.

```typescript
interface CustomContext extends ExpressContext {
{
    //Custom properties.
}

const contextFn = (context: ExpressContext):CustomContext => {
    return Object.assign({<Custom Properties>}, context);
}
```

Further passing the context down a chain of resolvers possibly extending the `CustomContext` typings even further.

```typescript

interface UserContext extends CustomContext {
    user: User,
}

//Assume function is part of a resolver chain.
const resolverFn = async (src: SomeObject, args: SomeArguments, context: UserContext) => {
    context.user = await context.models.User.find({uName: args.uName});
}
```

It's a minor inconvenience not having the typings exported since that means that i have to define the context typings my self and possibly keep an eye out for future changes. Or import the typings as a submodule and possibly loose dependency if there's a sudden decision to refactor and move files around.

**Comment by eliw00d** [comment]
@abernix Here is another example of usage that I came across, if it helps:
```typescript
import { ExpressContext } from 'apollo-server-express/src/ApolloServer'

const gateway = new ApolloGateway({
  buildService: ({ url }) => new RemoteGraphQLDataSource<ExpressContext>({
    ...
  })
})
```

**Comment by shellscape** [comment]
Seconding this. It's extremely dirty to have to import from source. (It's dubious that Apollo is choosing to export it's source in the first place)

**Comment by abernix** [comment]
Closed by #4821.

**Comment by shellscape** [comment]
Thanks!
